src/project.rs

```diff
@@ -572,6 +577,105 @@ impl FuzzProject {
         Ok(())
     }
 
+    /// Produce coverage information for a given corpus
+    pub fn exec_coverage(self, coverage: &options::Coverage) -> Result<()> {
+        // Build project with source-based coverage generation enabled
+        self.exec_build(&coverage.build, Some(&coverage.target))?;
+
+        // Retrieve corpus directories
+        let corpora = if coverage.corpus.is_empty() {
+            vec![self.corpus_for(&coverage.target)?]
+        } else {
+            coverage
+                .corpus
+                .iter()
+                .map(|name| Path::new(name).to_path_buf())
+                .collect()
+        };
+
+        // Collect the (non-directory) readable input files from the corpora
+        let files_and_dirs = corpora.iter().flat_map(fs::read_dir).flatten().flatten();
+        let mut readable_input_files = files_and_dirs
+            .filter(|file| match file.file_type() {
+                Ok(ft) => ft.is_file(),
+                _ => false,
+            })
+            .peekable();
+        if readable_input_files.peek().is_none() {
+            bail!(
+                "The corpus does not contain program-input files. \
+                 Coverage information requires existing input files. \
+                 Try running the fuzzer first (`cargo fuzz run ...`) to generate a corpus, \
+                 or provide a nonempty corpus directory."
+            )
+        }
+
+        let (coverage_out_raw_dir, coverage_out_file) = self.coverage_for(&coverage.target)?;
+
+        // Generating individual coverage data for all files in corpora
+        for input_file in readable_input_files {
+            let (mut cmd, file_name) =
+                self.create_coverage_cmd(coverage, &coverage_out_raw_dir, &input_file.path())?;
+            eprintln!("Generating coverage data for {:?}", file_name);
+            cmd.output()
```

Comment:
Calling `.output()` and then not using the output is a bit inefficient. `.output()` will buffer up stdout and stderr and return them, but if we don't actually care about the output, and we also don't want it printed to *our* stderr/stdout, then we can do 

```rust
cmd
    .stdout(std::process::Stdio::null())
    .stderr(std::process::Stdio::null())
```

when constructing the command, and then run the command via

```rust
let status = cmd.status().with_context(|| ...)?;
anyhow::ensure!(status.success, "command exited with failure code {}: {:?}", status.code, cmd);
```

(Note: we also want to check that the command exited successfully, as the above snippet does)

Reply:
I actually do want to preserve the output of those commands, especially if there is a failure. So I am just using `cmd.status...` similarly to what you suggested.